src/app/import/components/upload/upload.component.ts

```diff
@@ -1,184 +1,162 @@
+    }
   }
 
-  reSubscribeToNotifications() {
-
-    NotificationService.unsubscribeFromNotification(this, null);
-    NotificationService.subscribeToNotification(this, {
-      projectId: this.projectId,
-      whitelist: ['file_upload'],
-      func: this.handleWebsocketNotification
-    });
-
+  ngOnDestroy(): void {
+    if (this.uploadTask$) this.uploadTask$.unsubscribe();
+    NotificationService.unsubscribeFromNotification(this, this.projectId)
   }
 
-  onFileInput(files: FileList | null): void {
-    if (files) {
-      this.file = files.item(0);
-      this.fileAttached.emit(this.file);
-      this.fileInput.nativeElement.value = '';
-    }
+  onFileDropped(files: File[]): void {
+    this.file = files.length > 0 ? files[0] : null;
+    this.fileAttached.emit(this.file);
   }
 
-  onSubmit(importOptions: string) {
-    if (this.file) {
-      let filename = this.file.name
-      if (this.uploadFileType.value == 'records') {
-        filename = this.getFinalFileName(filename);
-      }
-      let tasks$ = [];
-      if (this.init && this.projectId == null) tasks$.push(this.createEmptyProject());
-
-      if (tasks$.length != 0) {
-        //wait project creation
-        forkJoin([...tasks$])
-          .pipe(first())
-          .subscribe((results: any) => {
-            this.projectId = results[0].id;
-            this.prepareEmptyProject();
-            this.finishUpUpload(filename, importOptions);
-          });
-      } else {
-        this.finishUpUpload(filename, importOptions);
-      }
-    }
+  onFileInput(event: any): void {
+    event.stopPropagation();
+    this.onFileDropped(event.target.files);
+    this.fileUpload.nativeElement.value = '';
   }
 
-  finishUpUpload(filename, importOptions) {
-    let tasks$ = [];
-    tasks$.push(this.getUploadCredentialsAndId(filename, importOptions));
-
-    forkJoin([...tasks$])
-      .pipe(first())
-      .subscribe((results) => {
-        this.uploadFile(results[0], filename)
-      });
+  onFileRemove(event: Event): void {
+    event.stopPropagation();
+    this.onFileDropped([]);
+    this.fileUpload.nativeElement.value = '';
   }
 
-  prepareEmptyProject() {
-    this.projectApolloService
-      .changeProjectTokenizer(this.projectId, 'en_core_web_sm')
-      .pipe(first()).subscribe();
-    this.projectApolloService
-      .updateProjectStatus(
-        this.projectId,
-        ProjectStatus.INIT_COMPLETE
-      ).pipe(first()).subscribe();
+  createProject(): Observable<any> {
+    return this.projectApolloService
+      .createProject("Imported Project", "Created during file upload " + this.file.name).pipe(first());
   }
 
-  createEmptyProject() {
+  updateTokenizerAndProjectStatus(projectId: string): void {
+    this.projectApolloService.changeProjectTokenizer(projectId, this.recordNewUploadHelper.selectedTokenizer).pipe(first()).subscribe();
+    this.projectApolloService.updateProjectStatus(projectId, ProjectStatus.INIT_COMPLETE).pipe(first()).subscribe();
+  }
```

Comment:
i would expect this to be in the upload helper class -> its also only called from there so why not make it a private function in the class?